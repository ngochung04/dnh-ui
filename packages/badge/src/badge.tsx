import React, { FC } from "react";
interface Props {
  children: React.ReactNode;
  variant?: "primary" | "danger";
}
export const Badge = ({
  children,
  variant,
  ...props
}: Props & React.HTMLAttributes<HTMLSpanElement>) => {
  const primary = {
    backgroundColor: "black",
    color: "blue",
    border: "1px solid blue",
    padding: "10px 20px",
    borderRadius: "5px",
    cursor: "pointer",
    fontSize: "1.2em",
    fontWeight: "bold",
    outline: "none",
    transition: "all 0.2s ease-in-out",
  };
  const danger = {
    backgroundColor: "black",
    color: "red",
    border: "1px solid red",
    padding: "10px 20px",
    borderRadius: "5px",
    cursor: "pointer",
    fontSize: "1.2em",
    fontWeight: "bold",
    outline: "none",
    transition: "all 0.2s ease-in-out",
  };

  const white = {
    backgroundColor: "black",
    color: "white",
    border: "1px solid white",
    padding: "2px 10px",
    borderRadius: "10px",
    cursor: "pointer",
    fontSize: "0.5em",
    fontWeight: "nomarl",
    outline: "none",
    transition: "all 0.2s ease-in-out",
  };

  const style = (variant: string) => {
    switch (variant) {
      case "primary":
        return primary;
      case "danger":
        return danger;
      default:
        return white;
    }
  };
  return (
    <span style={style(variant ?? "")} {...props}>
      {children}
    </span>
  );
};
