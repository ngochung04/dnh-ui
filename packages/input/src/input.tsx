import React from "react";
interface Props {}
export const Input = ({
  children,
  ...props
}: Props & React.InputHTMLAttributes<HTMLInputElement>) => {
  return <input {...props}>{children}</input>;
};
