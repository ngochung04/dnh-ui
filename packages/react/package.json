{
  "name": "@dnh-ui/react",
  "version": "0.1.2",
  "description": "description",
  "main": "dist/dnh-ui-react.cjs.js",
  "module": "dist/dnh-ui-react.esm.js",
  "types": "dist/dnh-ui-react.cjs.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ngochung04/dnh-ui.git",
    "directory": "packages/react"
  },
  "dependencies": {
    "@dnh-ui/button": "latest",
    "@dnh-ui/input": "latest",
    "@dnh-ui/badge": "latest"
  }
}