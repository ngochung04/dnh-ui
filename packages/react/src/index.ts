export * from "@dnh-ui/button";
export * from "@dnh-ui/input";
export * from "@dnh-ui/badge";
