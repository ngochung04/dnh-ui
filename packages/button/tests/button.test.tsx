import * as React from "react";
import { Button } from "../src";
import { render } from "enzyme";
import "jest";

describe("Button", () => {
  it("Button render", () => {
    const component = render(<Button>abc</Button>);
    expect(component).toMatchSnapshot();
  });

  it("Button children", () => {
    const component = render(<Button>abc</Button>);
    expect(component.text()).toBe("abc");
  });
});
