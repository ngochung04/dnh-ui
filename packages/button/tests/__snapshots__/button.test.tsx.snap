// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button render 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "style": "background-color:black;color:white;border:1px solid white;padding:10px 20px;border-radius:5px;cursor:pointer;font-size:1.2em;font-weight:bold;outline:none;transition:all 0.2s ease-in-out",
    },
    "children": Array [
      Node {
        "data": "abc",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
